openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py

```diff
@@ -0,0 +1,173 @@
+
+    def test_sending_statistics(self, mock_request):
+        """
+        Verify that send_instance_statistics_to_acceptor sends request to acceptor API for token dispatch statistics.
+        """
+        send_instance_statistics_to_acceptor('https://mock-url.com', {'mock_data': 'mock_data', })
+
+        mock_request.assert_called_once_with(
+            'https://mock-url.com' + '/api/installation/statistics/', {'mock_data': 'mock_data', }
+        )
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.utils.logging.Logger.info')
+    def test_successfully_sent_statistics(self, mock_logging, mock_request):
+        """
+        Verify that send_instance_statistics_to_acceptor successfully dispatches
+        installation_statistics to acceptor API.
+        """
+        mock_request.return_value.status_code = 201
+
+        send_instance_statistics_to_acceptor('https://mock-url.com', {'mock_data': 'mock_data', })
+
+        mock_logging.assert_called_with('Data were successfully transferred to OLGA acceptor. Status code is 201.')
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.utils.logging.Logger.info')
+    def test_unsuccessfully_sent_statistics(self, mock_logging, mock_request):
+        """
+        Verify that send_instance_statistics_to_acceptor unsuccessfully dispatches
+        installation_statistics to acceptor API.
+        """
+        mock_request.return_value.status_code = 400
+
+        send_instance_statistics_to_acceptor('https://mock-url.com', {'mock_data': 'mock_data', })
+
+        mock_logging.assert_called_with(
+            'Data were not successfully transferred to OLGA acceptor. Status code is {0}.'.format(
+                mock_request.return_value.status_code
+            )
+        )
+
+
+class TestDispatchInstallationStatisticsAccessToken(unittest.TestCase):
+    """
+    Tests for dispatching installation statistics access token.
+    """
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.token_utils.get_access_token')
+    def test_get_access_token_occurs(self, mock_get_access_token):
+        """
+        Verify that get_acceptor_api_access_token method calls get_access_token method.
+        """
+        get_acceptor_api_access_token('mock-url.com')
+
+        mock_get_access_token.assert_called_once()
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.token_utils.access_token_registration')
+    @patch('openedx.core.djangoapps.edx_global_analytics.token_utils.get_access_token')
+    def test_token_registration_occurs(self, mock_get_access_token, mock_access_token_registration):
+        """
+        Verify that get_acceptor_api_access_token method calls access_token_registration
+        if access token does not exists.
+        """
+        mock_get_access_token.return_value = ''
+
+        get_acceptor_api_access_token('mock-url.com')
+
+        mock_access_token_registration.assert_called_once_with('mock-url.com')
```

Comment:
Tests in general are good as whole, but they need to be restructured. Lets discuss.